test/test_esmre.py

```diff
@@ -22,136 +23,140 @@
+        self.checkHints({"bristol"}, r"bristol *fasion")
 
     def testIncludesRepeatedCharacterButStartsNewHint(self):
-        self.checkHints(["ava", "st me harties"], r"ava+st me harties")
+        self.checkHints({"st me harties"},
+                         r"ava+st me harties")
 
     def testSkipsGroupsWithAlternation(self):
-        self.checkHints(
-            ["Hoist the ", ", ye ", "!"],
-            r"Hoist the (mizzen mast|main brace), "
-            r"ye (landlubbers|scurvy dogs)!",
-        )
+        self.checkHints({"hoist the "},
+             r"Hoist the (mizzen mast|main brace), "
+                       r"ye (landlubbers|scurvy dogs)!")
 
     def testSkipsAny(self):
-        self.checkHints(
-            ["After 10 paces, ", " marks the spot"],
-            r"After 10 paces, . marks the spot",
-        )
+        self.checkHints({"after 10 paces, "},
+                         r"After 10 paces, . marks the spot")
 
     def testSkipsOneOrMoreAny(self):
-        self.checkHints(["Hard to ", "!"], r"Hard to .+!")
+        self.checkHints({"hard to "},
+                         r"Hard to .+!")
 
     def testSkipsNestedGroups(self):
-        self.checkHints(
-            ["Squark!", " Pieces of ", "!"], r"Squark!( Pieces of (.+)!)"
-        )
+        self.checkHints({" pieces of "},
+                         r"Squark!( Pieces of (.+)!)")
 
     def testSkipsCharacterClass(self):
-        self.checkHints(["r"], r"[ya]a*r+")
+        self.checkHints({"r"},
+                         r"[ya]a*r+")
 
     def testRightBracketDoesNotCloseGroupIfInClass(self):
-        self.checkHints([":=", "X"], r":=([)D])X")
+        self.checkHints({":="},
+                         r":=([)D])X")
 
     def testSkipsBackslashMetacharacters(self):
-        self.checkHints(["Cap'n", " ", " Beard"], r"Cap'n\b ([\S] Beard)")
+        self.checkHints({" beard"},
+                         r"Cap'n\b ([\S] Beard)")
 
     def testBackslashBracketDoesNotCloseGroup(self):
-        self.checkHints([":=", "X"], r":=(\)|D)X")
+        self.checkHints({":="},
+                         r":=(\)|D)X")
 
     def testBackslashSquareBracketDoesNotCloseClass(self):
-        self.checkHints([":=", "X"], r":=[)D\]]X")
+        self.checkHints({":="},
+                         r":=[)D\]]X")
 
     def testSkipsMetacharactersAfterGroups(self):
-        self.checkHints(
-            ["Yo ", "ho ", " and a bottle of rum"],
-            r"Yo (ho )+ and a bottle of rum",
-        )
+        self.checkHints({" and a bottle of rum"},
+                        r"Yo (ho )+ and a bottle of rum")
 
     def testSkipsRepetionBraces(self):
-        self.checkHints(["A", ", me harties"], r"Ar{2-10}, me harties")
+        self.checkHints({", me harties"},
+                        r"Ar{2-10}, me harties")
 
-    def testAlternationCausesEmptyResult(self):
-        self.checkHints([], r"rum|grog")
+    def testAlternation(self):
+        self.checkHints({"rum", "grog"}, r"rum|grog")
```

Comment:
This is one of the main improvements — an alternation can now make two index entries.